pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1536,7 +1536,7 @@ func (m *kubeGenericRuntimeManager) GetPodStatus(ctx context.Context, uid kubety
 
 	sandboxStatuses := []*runtimeapi.PodSandboxStatus{}
 	containerStatuses := []*kubecontainer.Status{}
-	var timestamp time.Time
+	timestamp := time.Now()
```

Comment:
what the story here? How it worked for non-Evented PLEG? I don't see this field initialized outside the evented pleg feature gate

Reply:
This timestamp variable was introduced in the EventedPLEG feature [PR#111318](https://github.com/kubernetes/kubernetes/pull/111384/files#diff-0e1a8d22b0f1dc42d412749c6a916b4c76394ead739d4bfbf26564faedd74a2f) to address timestamp race conditions when updating pod status cache. It should ideally correspond to the time the container runtime retrieves the pod status and should be returned through the CRI `PodSandboxStatus()`. When using generic pleg, this variable is not used, so it does not affect the existing non-EventedPLEG logic. However, when EventedPLEG is enabled and the CRI `PodSandboxStatus()` does not return a timestamp (as with containerd, which I believe should also return it to reduce the overhead of an additional request), timestamp in PodStatus returned by `GetPodStatus()` will always be zero. This means that the pod status cache will never be updated when running a generic pleg Relist() to call `updateCache()`, because timestamp overwritten  to zero is impossible newer than exist cached timestamp. In this scenario, running the generic pleg at longer intervals(5 minutes) is futile and meaningless when EventePLEG is enabled. So I changed the zero timestamp to now for making it possible to update the cache.

Reply:
This timestamp variable was introduced in the EventedPLEG feature [PR#111318](https://github.com/kubernetes/kubernetes/pull/111384/files#diff-0e1a8d22b0f1dc42d412749c6a916b4c76394ead739d4bfbf26564faedd74a2f) to address timestamp race conditions when updating pod status cache. It should ideally correspond to the time the container runtime retrieves the pod status and should be returned through the CRI `PodSandboxStatus()`. When using generic pleg, this variable is not used, so it does not affect the existing non-EventedPLEG logic. However, when EventedPLEG is enabled and the CRI `PodSandboxStatus()` does not return a timestamp (as with containerd, which I believe should also return it to reduce the overhead of an additional request), timestamp in PodStatus returned by `GetPodStatus()` will always be zero. This means that the pod status cache will never be updated when running a generic pleg Relist() to call `updateCache()`, because timestamp overwritten  to zero is impossible newer than exist cached timestamp. In this scenario, running the generic pleg at longer intervals(5 minutes) is futile and meaningless when EventePLEG is enabled. So I changed the zero timestamp to now for making it possible to update the cache.

Reply:
I see that timestamp is initialized here and then used in a return object unconditionally. Is it not being used for generic PLEG? Any sideeffects for the generic pleg?

Reply:
The timestamp is only used in updateCache(), ref: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/pleg/generic.go#L483.
```go
	if utilfeature.DefaultFeatureGate.Enabled(features.EventedPLEG) && isEventedPLEGInUse() && status != nil {
		timestamp = status.TimeStamp
	}
``` 
 I also checked all other places where this function `GetPodStatus` called and none of them use this timestamp field. I believe this changed is safe for generic pleg.